examples/chip-tool/commands/group/Commands.h

```diff
@@ -0,0 +1,384 @@
+        auto iter                                                = groupDataProvider->IterateGroupKeys(fabricIndex);
+        size_t maxCount                                          = iter->Count();
+        chip::Credentials::GroupDataProvider::GroupKey groupKey;
+        while (iter->Next(groupKey))
+        {
+            if (groupKey.group_id == groupId && groupKey.keyset_id == keysetId)
+            {
+                break;
+            }
+            index++;
+        }
+        iter->Release();
+        if (index >= maxCount)
+        {
+            err = CHIP_ERROR_INTERNAL;
+        }
+        else
+        {
+            err = groupDataProvider->RemoveGroupKeyAt(fabricIndex, index);
+        }
+
+        SetCommandExitStatus(err);
+        return err;
+    }
+
+private:
+    chip::GroupId groupId;
+    chip::KeysetId keysetId;
+};
+
+class AddKeySet : public CHIPCommand
+{
+public:
+    AddKeySet(CredentialIssuerCommands * credsIssuerConfig) : CHIPCommand("add-keyset", credsIssuerConfig)
+    {
+        AddArgument("keysetId", 0, UINT16_MAX, &keysetId);
+        AddArgument("keyPolicy", 0, UINT16_MAX, &keyPolicy);
+        AddArgument("EpochKey", &epochKey);
+    }
+    chip::System::Clock::Timeout GetWaitDuration() const override { return chip::System::Clock::Seconds16(5); }
+
+    CHIP_ERROR RunCommand() override
+    {
+        CHIP_ERROR err = CHIP_NO_ERROR;
+        chip::FabricIndex fabricIndex;
+        CurrentCommissioner().GetFabricIndex(&fabricIndex);
+        chip::Credentials::GroupDataProvider * groupDataProvider = chip::Credentials::GetGroupDataProvider();
+        uint8_t compressed_fabric_id[sizeof(uint64_t)];
+        chip::MutableByteSpan compressed_fabric_id_span(compressed_fabric_id);
+        ReturnLogErrorOnFailure(CurrentCommissioner().GetFabricInfo()->GetCompressedId(compressed_fabric_id_span));
+        static const uint8_t epochKeySize = static_cast<uint8_t>(sizeof(chip::Credentials::GroupDataProvider::EpochKey));
+
+        if ((keyPolicy != chip::Credentials::GroupDataProvider::SecurityPolicy::kCacheAndSync &&
+             keyPolicy != chip::Credentials::GroupDataProvider::SecurityPolicy::kTrustFirst) ||
+            (epochKey.size() / 2) != epochKeySize)
+        {
+            err = CHIP_ERROR_INVALID_ARGUMENT;
+        }
+        else
+        {
+            chip::Credentials::GroupDataProvider::KeySet keySet(keysetId, keyPolicy, 1);
+            std::string str = std::string(epochKey.begin(), epochKey.end());
+            chip::Credentials::GroupDataProvider::EpochKey epoch_keys;
+            epoch_keys.start_time = static_cast<uint64_t>(std::stoi(str.substr(0, 16).c_str(), 0, 16));
+
+            for (uint32_t i = 16; i < ((epochKeySize * 2)); i += 2)
+            {
+                epoch_keys.key[(i - 16) / 2] = static_cast<uint8_t>(std::stoi(str.substr(i, 2).c_str(), 0, 16));
+            }
```

Comment:
Why not just have the docs say to prefix the string with `hex:` and then the conversion from hex to binary will be done automatically and the ByteSpan can be used as-is, as raw bytes.

Reply:
Wasn't aware of that feature, will do so.